src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1319,56 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        macro_obj = macro_obj
+
+        if macro_obj.getParentMacro() is None:
+            try:
+                logging_onoff = macro_obj.getEnv("LogMacro")
+            except:
+                logging_onoff = False
+            if logging_onoff:
+                try:
+                    self.logging_mode = macro_obj.getEnv("LogMacroMode")
+                except:
+                    self.logging_mode = 0
+                try:
+                    logging_path = macro_obj.getEnv("LogMacroDir")
+                except:
+                    macro_obj.setEnv("LogMacroDir", "/tmp")
+                    logging_path = macro_obj.getEnv("LogMacroDir")
+                    
+                log_file = logging_path + "/spock_session_" + socket.gethostname() +".log"
+
+                if self.logging_mode:
+                    if os.path.isfile(log_file):
+                        try:
+                            os.system("vrsn -s %s" % log_file)
```

Comment:
What `vrsn` does? It does not look like a platform independent thing.

It must to do something with the append/new file feature. What about using the `RotatingFileHandler` from python logging?

Reply:
As far as I could read and test, the RotatingFIleHandler allows the rotation of the files based on the size, but what the users want is to have one file for each macro (in case they run scans, so that each scan is in one file). I do not get
this with a RotatingFileHandler.

Reply:
I've never used it this way so it may not work. But please try to set the `maxBytes=0` or `backupCount=0` in the constructor (I think that these are even the default values) and then call the `doRollover` when you wan to change the file. This is how I interpret the [docs](https://docs.python.org/2/library/logging.handlers.html#rotatingfilehandler).

BTW why the users want to have a dedicated log file per scan? Isn't the role of the `ScanFile`?

Reply:
Hi @reszelaz, I found the way to do this. It is commited.

Reply:
Great that it works! I have tested it and the current behavior is that as long as the mode is set to 1 (using the `logmacro` macro) every macro execution starts a new file. Is it correct? Then to simply change the file one would need to execute the `logmacro` macro twice, isn't it? Like for example:
```
logmacro True 1
logmacro True 0
```

Reply:
Yes, it works exactly like you describe. Thanks for testing.

Reply:
Thanks, it looks good! Just an idea, no need to change it:
What about having a specific macro `logmacro_rollover` that would do this? Then, I think that we could remove the `mode` argument of the `logmacro`.

Reply:
Do you mean with an argument to set it on or off?

Reply:
Maybe I did not understand well this feature. Are the users also interested in producing one log file per one macro execution. Or, in other words, would they continue executing macros when the mode is 1?
If not, and this is just to make a rollowever and then return to the mode 0, I think that the parameter is not necessary. The macro execution would do the rollover and that's it.

Reply:
They requested to have it automatically, per one macro execution.

Reply:
Aaa, ok. Then forget about my proposal. I understood it wrong. THanks!